sip-ri/src/main/java/gov/nist/javax/sip/header/extensions/Diversion.java

```diff
@@ -0,0 +1,363 @@
+   }
+   
+   public String getDiversionAddress() {
+	   return diversionAddress;
+   }
+   
+   /**
+    * Gets the Reason parameter of the Diversion header.
+    *
+    * @return the remote tag value
+    */
+   public String getReason() {
+   	 if (parameters == null)
+            return null;
+        return getParameter(ParameterNames.REASON);
+   }
+   /**
+    * Sets the remote tag of the Target-Dialog header.
+    *
+    * @param remoteTag the remote tag value to set
+    * @throws ParseException if the provided remoteTag is invalid
+    */
+   public void setReason(String r) throws ParseException {
+   	  if (r == null)
+             throw new NullPointerException("null param");
+         else if (r.trim().equals(""))
+             throw new ParseException("bad param", 0);
+         this.setParameter(ParameterNames.REASON, r);
+   }
+   /** Boolean function
+    * @return true if the parameter exist
+    */
+   public boolean hasReason() {
+       return hasParameter(ParameterNames.REASON);
+   }
+
+   /** remove Tag member
+    */
+   public void removeReason() {
+       parameters.delete(ParameterNames.REASON);      
+   }
+   
+   /**
+    * Gets the diversion-limit parameter from the address param list.
+    * @return field
+    */
+   public String getLimit() {
+       if (parameters == null)
+           return null;
+       return getParameter(ParameterNames.LIMIT);
+   }
+   
+   /**
+    * Sets the diversion-limit parameter of the Diversion header.
+    *
+    * @param reason diversion-limit parameter value to set
+    * @throws ParseException if the provided parameter is invalid
+    */
+   public void setLimit(String l) throws ParseException {
+   	  if (l == null)
+             throw new NullPointerException("no limit param");
+         else if (l.trim().equals(""))
+             throw new ParseException("bad parameter", 0);
+         this.setParameter(ParameterNames.LIMIT, l);
+   }
+   
+   /** Boolean function
+    * @return true if the parameter exist
+    */
+   public boolean hasLimit() {
+       return hasParameter(ParameterNames.LIMIT);
+   }
+   
+   /** remove diversion-limit member
+    */
+   public void removeLimit() {
+       parameters.delete(ParameterNames.LIMIT);
+   }
+   
+   
+   /**
+    * Gets the diversion-counter parameter from the address param list.
+    * @return field
+    */
+   public String getCounter() {
+       if (parameters == null)
+           return null;
+       return getParameter(ParameterNames.COUNTER);
+   }
+   
+   /**
+    * Sets the diversion-counter parameter of the Diversion header.
+    *
+    * @param reason diversion-counter parameter value to set
+    * @throws ParseException if the provided parameter is invalid
+    */
+   public void setCounter(String c) throws ParseException {
+   	  if (c == null)
+             throw new NullPointerException("no counter param");
+         else if (c.trim().equals(""))
+             throw new ParseException("bad parameter", 0);
```

Comment:
Maybe the parameter of that method shouldn't be a String or we can add a check to ensure it match the syntax of 2DIGIT at https://datatracker.ietf.org/doc/html/rfc5806#section-4 i.e: diversion-counter = "counter" "=" 1*2DIGIT